From 60e1f652eed7630ecaf891b4640ff929dd4d3389 Mon Sep 17 00:00:00 2001
From: Ian Glen <ian.glen@boxcast.com>
Date: Mon, 27 Dec 2021 14:38:47 -0500
Subject: [PATCH] Use multiprocessing instead of threading

---
 gpio/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gpio/__init__.py b/gpio/__init__.py
index c6f37f7..abb440f 100644
--- a/gpio/__init__.py
+++ b/gpio/__init__.py
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 __version__ = '1.0.0'
 
-from threading import Lock
+from multiprocessing import Lock
 try:
     from collections.abc import Iterable
 except ImportError:
