From 88f94a4757ac8da9eeba961616aed7939e41e4f1 Mon Sep 17 00:00:00 2001
From: OpenEmbedded <oe.patch@oe>
Date: Fri, 24 Dec 2021 22:27:56 -0500
Subject: [PATCH] Adjust ILI9481 power settings

---
 drivers/staging/fbtft/fb_ili9481.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/fbtft/fb_ili9481.c b/drivers/staging/fbtft/fb_ili9481.c
index 34832feebbd0..79dd9ffd9080 100644
--- a/drivers/staging/fbtft/fb_ili9481.c
+++ b/drivers/staging/fbtft/fb_ili9481.c
@@ -24,10 +24,13 @@ static const s16 default_init_sequence[] = {
 	-1, MIPI_DCS_EXIT_SLEEP_MODE,
 	-2, 50,
 	/* Power setting */
-	-1, 0xD0, 0x07, 0x42, 0x18,
+	/* VC=1, BT=1, PON=1, VREG1OUT=4.000V, VCIRE=1 */
+	-1, 0xD0, 0x01, 0x41, 0x18,
 	/* VCOM */
+	/* SELVCM=0, VCM=7, VDV=10 */
 	-1, 0xD1, 0x00, 0x07, 0x10,
 	/* Power setting for norm. mode */
+	/* AP0=1, DC00=2, DC10=0 */
 	-1, 0xD2, 0x01, 0x02,
 	/* Panel driving setting */
 	-1, 0xC0, 0x10, 0x3B, 0x00, 0x02, 0x11,
